lib/bot.js

```diff
@@ -129,9 +129,11 @@ class Bot {
       channels.forEach((channelName) => {
         const channel = channelName.toLowerCase();
         if (this.channelUsers[channel]) {
-          this.channelUsers[channel].delete(oldNick);
-          this.channelUsers[channel].add(newNick);
-          this.sendExactToDiscord(channel, `*${oldNick}* is now known as ${newNick}`);
+          if (this.channelUsers[channel].has(oldNick)) {
+            this.channelUsers[channel].delete(oldNick);
+            this.channelUsers[channel].add(newNick);
+            this.sendExactToDiscord(channel, `*${oldNick}* is now known as ${newNick}`);
+          }
         } else {
           logger.warn(`No channelUsers found for ${channel} when ${oldNick} changed.`);
```

Comment:
The logger warning probably isn't necessary – it was mostly to get around the issue with lowercasing channel names, in the other events.